streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -345,8 +343,13 @@ private SinkNodeFactory(final String name,
         }
     }
 
+    public void setTopologyName(final String namedTopology) {
```

Comment:
Oh man, this was a whole thing. No we definitely don't want to be able to rename them, but due to various java shenanigans it's pretty much impossible to get this name passed in to the constructor (it comes down to being unable to set `this.topologyName = topologyName` before calling `super();` in the `NamedTopologyStreamsBuilder` constructor, since its superclass directly instantiates the InternalTopologyBuilder. However, I'll add a check and make sure it's not possible to accidentally re-set the name once it's been set

Reply:
I think the check should be sufficient